### ISSUE TITLE
libheif 1.19.1 breaks build

### ISSUE THREAD

**Issue by 0-wiz-0** [bigcat88/pillow_heif#306]
### Describe the bug

libheif 1.19.1 is out, and building pillow_heif-0.20.0 against it fails with:
```
gcc -Wsign-compare -DNDEBUG -O2 -g -fstack-clash-protection -I/usr/include -I/usr/include/readline -I/usr/pkg/include -O2 -g -fstack-clash-protection -I/usr/include -I/usr/include/readline -I/usr/pkg/include -O2 -g -fstack-clash-protection -I/usr/include -I/usr/include/readline -I/usr/pkg/include -O2 -g -fstack-clash-protection -I/usr/include -I/usr/pkg/include/python3.11 -I/usr/pkg/include -g -I/usr/include -I/usr/pkg/include/python3.11 -I/usr/pkg/include -fPIC -I/usr/include -I/usr/pkg/include/python3.11 -I/usr/pkg/include -I/usr/pkg/include/python3.11 -c pillow_heif/_pillow_heif.c -o build/temp.netbsd-10.99.12-amd64-cpython-311/pillow_heif/_pillow_heif.o -Ofast -Werror
pillow_heif/_pillow_heif.c: In function '_CtxWriteImage_encode':
pillow_heif/_pillow_heif.c:550:59: error: assignment of member 'color_primaries' in read-only object
  550 |             options->output_nclx_profile->color_primaries = color_primaries;
      |                                                           ^
pillow_heif/_pillow_heif.c:552:68: error: assignment of member 'transfer_characteristics' in read-only object
  552 |             options->output_nclx_profile->transfer_characteristics = transfer_characteristics;
      |                                                                    ^
pillow_heif/_pillow_heif.c:554:63: error: assignment of member 'matrix_coefficients' in read-only object
  554 |             options->output_nclx_profile->matrix_coefficients = matrix_coefficients;
      |                                                               ^
pillow_heif/_pillow_heif.c:556:59: error: assignment of member 'full_range_flag' in read-only object
  556 |             options->output_nclx_profile->full_range_flag = full_range_flag;
      |                                                           ^
pillow_heif/_pillow_heif.c:561:45: error: passing argument 1 of 'heif_nclx_color_profile_free' discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  561 |         heif_nclx_color_profile_free(options->output_nclx_profile);
      |                                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from pillow_heif/_pillow_heif.c:4:
/scratch/graphics/py-pillow_heif/work/.buildlink/include/libheif/heif.h:1627:67: note: expected 'struct heif_color_profile_nclx *' but argument is of type 'const struct heif_color_profile_nclx *'
 1627 | void heif_nclx_color_profile_free(struct heif_color_profile_nclx* nclx_profile);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
cc1: all warnings being treated as errors
```

### Steps/Code to Reproduce

Install libheif 1.19.1, built pillow_heif 0.20.0.

### Expected Results

Build succeeds.

### Actual Results

Build fails, see above.

### Versions

```shell
Python 3.13.0, NetBSD 10.99.12/amd64, gcc 12.4.0.
```

**Comment by bigcat88** [comment]
I will take a look at it, after `libheif 1.19.3` release

**Comment by farindk** [comment]
I fixed this in https://github.com/strukturag/libheif/commit/3dd7019ff579c038cba96353390cd41edfda927e.

I hoped that I could fix the constness in the API, but I'll probably have to work around that internally.
You can close this, it will be fixed in v1.19.3.

**Comment by bigcat88** [comment]
> I fixed this in [strukturag/libheif@3dd7019](https://github.com/strukturag/libheif/commit/3dd7019ff579c038cba96353390cd41edfda927e).
> 
> I hoped that I could fix the constness in the API, but I'll probably have to work around that internally. You can close this, it will be fixed in v1.19.3.

Wow! Thanks you a lot.
